### PR TITLE
Berry pouch

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/berry_pouch.c [hunk 1]
```diff
@@ -0,0 +1,1528 @@
...
+static void CB2_InitBerryPouch(void);
+static bool8 RunBerryPouchInit(void);
+static void AbortBerryPouchLoading(void);
+static void Task_AbortBerryPouchLoading_WaitFade(u8 taskId);
+static void BerryPouchInitBgs(void);
+static bool8 BerryPouchLoadGfx(void);
+static bool8 AllocateListMenuBuffers(void);
+static void SetUpListMenuTemplate(void);
+static void GetBerryNameAndIndexForMenu(u8 * dest, u16 itemId);
+static void BerryPouchMoveCursorFunc(s32 itemIndex, bool8 onInit, struct ListMenu *list);
```
Comment (jiangzhengwenjz) [inline src/berry_pouch.c]: Inconsistent space style for pointer. You usually add 1 space before and 1 after  `*`.

File: src/berry_pouch.c [hunk 2]
```diff
@@ -0,0 +1,1528 @@
...
+static void Task_NormalContextMenu_0(u8 taskId);
+static void Task_NormalContextMenu_1(u8 taskId);
+static void Task_BerryPouch_Use_0(u8 taskId);
+static void Task_BerryPouch_Toss_0(u8 taskId);
+static void Task_AskTossMultiple(u8 taskId);
+static void Task_TossNo(u8 taskId);
+static void Task_Toss_SelectMultiple(u8 taskId);
+static void Task_TossYes(u8 taskId);
+static void Task_WaitButtonThenTossBerries(u8 taskId);
+static void Task_BerryPouch_Give_0(u8 taskId);
```
Comment (jiangzhengwenjz) [inline src/berry_pouch.c]: Why do we need this `_0`(for multiple functions)?

File: src/berry_pouch.c (hunk 1 above)
Comment (PikalaxALT) [reply src/berry_pouch.c]: Good catch. Fixed other instances to adhere to the following convention:
- No space between `*` and `const`
- No space between `struct StructName` or `union UnionName` and `*`
- Space between numeric type name and `*`

File: src/berry_pouch.c (hunk 2 above)
Comment (PikalaxALT) [reply src/berry_pouch.c]: Was meant to be intermediate nomenclature. Resolved.

**Pull request merged**